src/Dotnet.Installer.Console/Commands/ListCommand.cs

```diff
@@ -111,4 +133,181 @@ string ComponentStatus(string key, int majorVersion, bool isEol)
             Environment.Exit(-1);
         }
     }
+
+    private async Task<Dictionary<string, DotnetVersion?>> GetComponentVersions(
+        List<Component> components,
+        uint timeoutInMilliseconds)
+    {
+        var componentVersions = new Dictionary<string, DotnetVersion?>();
+        if (components.Count == 0) return componentVersions;
+
+        // thread safety of adding a key to a dictionary is questionable, therefore we pre-populate all keys
+        foreach (var component in components)
+        {
+            componentVersions.Add(component.Key, null);
+        }
```

Comment:
**suggestion:** Have you tried [`ConcurrentDictionary<TKey, TValue>`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-8.0)?

Reply:
I thought about it, but in my opinion it will create more overhead than it's worth, because it will unnecessarily block every thread when setting a value for a key, even though there is no need for it.